src/account/PluginManager2.sol

```diff
@@ -75,24 +98,43 @@ abstract contract PluginManager2 {
         FunctionReference validationFunction,
         bytes4[] calldata selectors,
         bytes calldata uninstallData,
-        bytes calldata preValidationHookUninstallData
+        bytes calldata preValidationHookUninstallData,
+        bytes calldata permissionHookUninstallData
     ) internal {
         AccountStorage storage _storage = getAccountStorage();
 
         _storage.validationData[validationFunction].isDefault = false;
         _storage.validationData[validationFunction].isSignatureValidation = false;
 
-        bytes[] memory preValidationHookUninstallDatas = abi.decode(preValidationHookUninstallData, (bytes[]));
-
-        // Clear pre validation hooks
-        EnumerableSet.Bytes32Set storage preValidationHooks =
-            _storage.validationData[validationFunction].preValidationHooks;
-        while (preValidationHooks.length() > 0) {
-            FunctionReference preValidationFunction = toFunctionReference(preValidationHooks.at(0));
-            preValidationHooks.remove(toSetValue(preValidationFunction));
-            (address preValidationPlugin,) = FunctionReferenceLib.unpack(preValidationFunction);
-            if (preValidationHookUninstallDatas[0].length > 0) {
-                IPlugin(preValidationPlugin).onUninstall(preValidationHookUninstallDatas[0]);
+        {
+            bytes[] memory preValidationHookUninstallDatas = abi.decode(preValidationHookUninstallData, (bytes[]));
+
+            // Clear pre validation hooks
+            EnumerableSet.Bytes32Set storage preValidationHooks =
+                _storage.validationData[validationFunction].preValidationHooks;
+            while (preValidationHooks.length() > 0) {
+                FunctionReference preValidationFunction = toFunctionReference(preValidationHooks.at(0));
+                preValidationHooks.remove(toSetValue(preValidationFunction));
+                (address preValidationPlugin,) = FunctionReferenceLib.unpack(preValidationFunction);
+                if (preValidationHookUninstallDatas[0].length > 0) {
```

Comment:
I had a similar thought with the rest of this contract, but I want to flag here as well-- are we sure it's reasonable only to call `onIn/Uninstall()` when data length is nonzero? 

As a plugin dev, I may have to clear some data related to msg.sender for example. I think it makes it easier if we always call these functions.

Reply:
Good callout! Thinking through this, the balance here is between deciding between the 2 defaults:
1. [no array == empty call] - we potentially make unnecessary calls to plugins that don't need them
2. [no array == no call] - we potentially have to provide empty `byte("")`s  of calldata just to trigger the call

Given that most plugins should be stateful, and stateful plugins would always have a non-view onUninstall function, I'm in agreement that option 1 is the better default

Reply:
ooh hold up, I think I misunderstood something here.. I was thinking of "if `preValidationHookUninstallDatas.length == 0`, call onUninstall with `bytes("")` which is a reasonable optimization to make. But you meant `preValidationHookUninstallDatas[0].length == 0`, we should perform the call anyway, which is a must have. Will make suggested change and leave optimization out